### PR TITLE
Postman Collection Generator

### PR DESCRIPTION
**Pull request by gcatanese**
Create the Postman generator: from an OpenAPI file, it generates a Postman collection [v2.1.0](https://schema.postman.com/collection/json/v2.1.0/draft-07/collection.json) that can be imported in Postman to execute and test the requests.

The existing OpenAPI import in Postman is very basic (especially when using the API instead of the Postman GUI) resulting in a Collection that doesn't have all the examples and features of the OpenAPI file. When using the Postman GUI there are few options available but this approach is not suitable when automating the generation workflow (i.e. create a collection when a new OpenAPI is available).

This generator implements the generation of a Postman collection with the following:

- one request for each OpenAPI request example
- set the Authorization header
- create a Postman variable to set the API Key
- (Optionally) Postman variables for path variables
- (Optionally) Postman variables for placeholders found in the request body of the examples
- (Optionally) create Postman dynamic variables (i.e. `guid`, `isoTimestamp`)
- prettify JSON examples

I obviously volunteer to maintain the generator.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PostmanCollectionCodegen.java [hunk 1]
```diff
@@ -0,0 +1,839 @@
...
+    }
+
+    /**
+     * Returns human-friendly help for the generator.  Provide the consumer with help
+     * tips, parameters here
+     *
+     * @return A string value for the help message
+     */
+    public String getHelp() {
+        return "Generates a postman JSON file";
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PostmanCollectionCodegen.java:430]: Shall we mention the supported Postman collection format (v2.1.0) here?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PostmanCollectionCodegen.java [hunk 2]
```diff
@@ -0,0 +1,839 @@
...
+                addToList(codegenOperation);
+            }
+
+        }
+
+        return results;
+    }
+
+
+    void addToMap(CodegenOperation codegenOperation){
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PostmanCollectionCodegen.java:298]: can you add a docstring (1 line or 2) explaining what this function does?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PostmanCollectionCodegen.java (hunk 1 above)
Comment (gcatanese) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PostmanCollectionCodegen.java]: Done

**Comment by wing328** [comment]
thanks for the PR. let's give it a try.

**Comment by wing328** [comment]
Hi @gcatanese I did a test with
```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g postman-collection -i https://developer.allegro.pl/swagger.yaml -o /tmp/pc/
```
and got the following errors in the output (partial):

```
[main] WARN  o.o.c.l.PostmanCollectionCodegen - Error formatting JSON
com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'BINARY_DATA_HERE': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"BINARY_DATA_HERE"; line: 1, column: 17]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2418)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:759)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:3038)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2079)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:805)
	at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4759)
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3124)
	at org.openapitools.codegen.languages.PostmanCollectionCodegen.formatJson(PostmanCollectionCodegen.java:704)
	at org.openapitools.codegen.languages.PostmanCollectionCodegen.getPostmanRequests(PostmanCollectionCodegen.java:367)
	at org.openapitools.codegen.languages.PostmanCollectionCodegen.postProcessOperationsWithModels(PostmanCollectionCodegen.java:256)
	at org.openapitools.codegen.DefaultGenerator.processOperations(DefaultGenerator.java:1275)
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:609)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:953)
	at org.openapitools.codegen.cmd.Generate.execute(Generate.java:511)
	at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
[main] WARN  o.o.c.l.PostmanCollectionCodegen - Error formatting JSON
com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'BINARY_DATA_HERE': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"BINARY_DATA_HERE"; line: 1, column: 17]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2418)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:759)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:3038)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2079)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:805)
	at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4759)
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3124)
	at org.openapitools.codegen.languages.PostmanCollectionCodegen.formatJson(PostmanCollectionCodegen.java:704)
	at org.openapitools.codegen.languages.PostmanCollectionCodegen.getPostmanRequests(PostmanCollectionCodegen.java:367)
	at org.openapitools.codegen.languages.PostmanCollectionCodegen.postProcessOperationsWithModels(PostmanCollectionCodegen.java:256)
	at org.openapitools.codegen.DefaultGenerator.processOperations(DefaultGenerator.java:1275)
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:609)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:953)
	at org.openapitools.codegen.cmd.Generate.execute(Generate.java:511)
	at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
```

Can you please take a look when you've time?

**Pull request merged**